src/core/RecyclerListView.tsx

```diff
@@ -453,13 +459,34 @@ export default class RecyclerListView<P extends RecyclerListViewProps, S extends
             }
             this._refreshViewability();
         } else if (this.props.dataProvider !== newProps.dataProvider) {
+            const onStartReachedCalled = this._onStartReachedCalled;
             if (newProps.dataProvider.getSize() > this.props.dataProvider.getSize()) {
-                this._onEndReachedCalled = false;
+                this._setOnEdgeReachedCalled(false);
```

Comment:
We're caching the current value of `onStartReached` but it could change (to `false`) if we enter the next `if` block
Then the following expression can still pass

```js
if (viewabilityTracker && onStartReachedCalled) {
```

Can't we use `this._onStartReachedCalled` there?

Having to cache the value and not use the latest seems a bit fishy